src/nodes.coffee

```diff
@@ -475,6 +461,34 @@ exports.HoistTarget = class HoistTarget extends Base
   compileClosure: (o) ->
     @compileToFragments o
 
+#### Root
+
+# The root node of the node tree
+exports.Root = class Root extends Base
```

Comment:
This is so much cleaner than a `Block` sometimes being a root and sometimes not. This AST work is giving the node classes a much-needed refactor 👏